tests/integration/nupic/opf/htmpredictionmodel_serialization_test.py

```diff
@@ -224,28 +233,88 @@ def _runSimpleModelSerializationDeserialization(self, modelParams):
     # Rudimentary serialization/deserialization; flush out starting point for
     # tests
 
-    srcModel = ModelFactory.create(modelParams)
-    srcModel.enableInference({'predictedField': 'consumption'})
+    m1 = ModelFactory.create(modelParams)
+    m1.enableInference({'predictedField': 'consumption'})
     headers = ['timestamp', 'consumption']
 
     record = [datetime.datetime(2013, 12, 12), numpy.random.uniform(100)]
     modelInput = dict(zip(headers, record))
-    srcModel.run(modelInput)
+    m1.run(modelInput)
 
-    # Save and load after each batch. Clean up.
     # Serialize
     builderProto = HTMPredictionModelProto.new_message()
-    srcModel.write(builderProto)
+    m1.write(builderProto)
 
     # Construct HTMPredictionModelProto reader from populated builder
     readerProto = HTMPredictionModelProto.from_bytes(builderProto.to_bytes())
 
     # Deserialize
-    destModel = HTMPredictionModel.read(readerProto)
+    m2 = HTMPredictionModel.read(readerProto)
+    # Work around a serialization bug that doesn't save the enabled predicted
+    # field
+    # TODO this enableInference call should be removed after NUP-2449 is fixed.
+    m2.enableInference({'predictedField': 'consumption'})
+
+    # Run computes on m1 & m2 and compare results
+    record = [datetime.datetime(2013, 12, 14), numpy.random.uniform(100)]
+    modelInput = dict(zip(headers, record))
+    # NOTE: use deepcopy to guarantee no side-effect
+    r1 = m1.run(copy.deepcopy(modelInput))
+    r2 = m2.run(copy.deepcopy(modelInput))
+
+    #self.assertEqual(r1, r2)
```

Comment:
@scottpurdy, sorry for not making this clear - the new test `testPredictedFieldAndInferenceEnabledAreSaved` is the target of this PR. The commented-out statement in the other test method is part of work-in-progress that isn't working yet.